### PR TITLE
Fix the bottom sheet jumping outside the viewport when selecting a category 

### PR DESCRIPTION
**Pull request by andreeaceachir142**
# What 

- Fix the bottom sheet jumping outside the viewport when selecting a category 

# How 

- Add a new atom indicating when the bottom sheet has been loaded, which is set when the transition has ended on the `Sheet.jsx` component 
- Add a new atom indicating if there is a category property/URL parameter, and based on that, call the `scrollIntoView` on the `Categories.jsx` file. The scroll into view should only happen when the category has been set as a prop or URL parameter, and **not** when the user selects a category.

### CODE REVIEW

File: packages/map-template/src/components/Search/Categories/Categories.jsx [hunk 1]
```diff
@@ -136,14 +141,14 @@ function Categories({ onSetSize, getFilteredLocations, searchFieldRef }) {
...
 
     /*
-     * Use the active category state to scroll into view
-     * and center the category.
+     * If the active category is a prop/query parameter and the bottom sheet is loaded, 
+     * then scroll into view and center the active category. 
      */
     useEffect(() => {
-        if (activeCategory) {
+        if (activeCategory && hasCategoryProp && isBottomSheetLoaded) {
```
Comment (ammapspeople) [inline packages/map-template/src/components/Search/Categories/Categories.jsx:148]: Will this work on larger screens where the bottom sheet is not used?

Comment (andreeaceachir142) [reply packages/map-template/src/components/Search/Categories/Categories.jsx]: Good catch. I adjusted the check to also consider the desktop screens where there's no bottom sheet.

File: packages/map-template/src/atoms/hasCategoryPropState.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { atom } from 'recoil';
+
+const hasCategoryPropState = atom({
```
Comment (ammapspeople) [inline packages/map-template/src/atoms/hasCategoryPropState.js:3]: Would it be more clean to have the value from the prop as the value in the atom? We already have atoms for properties (locationId, startZoomLevel etc.), and this boolean state is kinda off pattern-wise.

Comment (andreeaceachir142) [reply packages/map-template/src/atoms/hasCategoryPropState.js]: Good idea. Refactored the category atom. ✅

**Review by ammapspeople — APPROVED** [review]
🆗

**Comment by andreeaceachir142** [comment]
A fix for the issue reported in Jira has been merged in the latest commit 🙏🏻 @ammapspeople

**Pull request merged**